src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -684,10 +684,18 @@ private async ValueTask<int> EnsureFullTlsFrameAsync<TIOAdapter>(CancellationTok
                 return frameSize;
             }
 
-            if (frameSize < int.MaxValue)
+            if (frameSize != int.MaxValue)
```

Comment:
Why is it needed? Looks like semantically it is the same. Only MaxValue can be "not <".